File: code/py/src/service_client/http/http_client_exception.py
"""Excepción lanzada cuando ocurre un error HTTP del lado del cliente (4xx) o servidor (5xx)."""

from __future__ import annotations


class HttpClientException(Exception):
    """Error HTTP con código de estado y cuerpo de respuesta opcional."""

    def __init__(
        self,
        *,
        status_code: int,
        message: str,
        response: dict[str, object] | None = None,
    ) -> None:
        super().__init__(message)
        self.status_code = status_code
        self.response = response

    def __str__(self) -> str:
        return f"HttpClientException: {self.args[0]} (status_code: {self.status_code})"

File: code/ts/src/http/http_client_exception.ts
/// Excepción lanzada cuando ocurre un error HTTP del lado del cliente (4xx) o servidor (5xx).
export class HttpClientException extends Error {
  readonly statusCode: number;
  readonly response?: Record<string, unknown>;

  constructor(opts: {
    statusCode: number;
    message: string;
    response?: Record<string, unknown>;
  }) {
    super(opts.message);
    this.name = 'HttpClientException';
    this.statusCode = opts.statusCode;
    this.response = opts.response;
  }

  override toString(): string {
    return `HttpClientException: ${this.message} (statusCode: ${this.statusCode})`;
  }
}
